dragonfly_doe2/doe/floor_space.py

```diff
@@ -0,0 +1,149 @@
+    activity: str = None
+    walls: List[Wall] = None
+
+    @classmethod
+    def from_room(cls, room: Room2D):
+        if not isinstance(room, Room2D):
+            # TODO: Make raise exception rather than print
+            print(
+                f'Unsupported Type: {type(room)}.\n'
+                'Expected dragonfly.room2d.Room2D'
+            )
+
+        wall_constr_name = room.properties.energy.construction_set.wall_set.exterior_construction.display_name
+        walls = []
+        bcs = [str(bc) for bc in room.boundary_conditions]
+        if 'Outdoors' in bcs:
+            for i, bc in enumerate(bcs):
+                if bc == 'Outdoors':
+                    walls.append(Wall().from_room_seg(
+                        room.display_name, i, wall_constr_name))
+        else:
+            walls = None
+
+        return cls(
+            name=room.display_name,
+            activity=room.properties.energy.program_type.display_name, walls=walls)
+
+    def to_inp(self):
+        space_walls = '\n'.join(wall.to_inp() for wall in self.walls)
+        space_block = f'"{self.name}" = SPACE\n' \
+                      f'   SHAPE           = POLYGON\n' \
+                      f'   POLYGON         = "{self.name} Plg"\n' \
+                      f'   C-ACTIVITY-DESC = *{self.activity}*\n   ..\n'
+        return space_block + space_walls
+
+    def __repr__(self):
+        return self.to_inp()
+
+
+@dataclass()
+class Floor:
+    """The *.inp 'Floor' object, contains spaces and space meta-data:
+    .. code-block::f#
+        "simple_example_dfb_Floor1" = FLOOR
+            Z                = 0.0
+            POLYGON          = "simple_example_dfb_Floor1 Floor Plg"
+            SHAPE            = POLYGON
+            FLOOR-HEIGHT     = 10.0
+            C-DIAGRAM-DATA   = *simple_example_dfb_Floor1 UI DiagData*
+            ..
+        "simple_example_dfb_Floor1_Room1" = SPACE
+            SHAPE            = POLYGON
+            POLYGON          = "simple_example_dfb_Floor1_Room1 Plg"
+            C-ACTIVITY-DESC  = *Generic Office Program*
+            ..
+        "simple_example_dfb_Floor1_Room1 Wall_1" = EXTERIOR-WALL
+            CONSTRUCTION    = "EWall Construction"
+            LOCATION        = SPACE-V1
+            ..
+    Object.
+    """
+    name: str = None
+    floor_z: float = None
+    floor_height: float = None
+    spaces: List[Space] = None
+
+    @classmethod
+    def from_story(cls, story: Story):
+        if not isinstance(story, Story):
+            # TODO: Make raise exception instead of print
+            print(f'Unsupported type: {type(story)}\n'
+                  'Expected dragonfy.story.Story')
+        cls_ = cls()
+        cls_.name = story.display_name
+        cls_.floor_z = story.floor_height
+        cls_.floor_height = story.floor_to_floor_height
+        spcs = []
+        for room in story.room_2ds:
+            spcs.append(Space().from_room(room))
+        cls_.spaces = spcs
+        return cls_
```

Comment:
Unless you need to access something from the class you should initiate it at the end. Here is how your code should look like:

```suggestion
        spaces = [Space.from_room(room) for room in story.room_2ds]
        return cls(story.display_name, story.floor_height, story.floor_to_floor_height, spaces)
```